Fhi.HelseId.Web/Services/HelseIdSecretHandlers.cs

```diff
@@ -197,38 +230,45 @@ public override void AddSecretConfiguration(IHelseIdWebKonfigurasjon configAuth,
             };
 #endif
         }
+
+        protected override JsonWebKey GetJsonWebKey()
+        {
+            return JsonWebKey;
+        }
     }
 
     public class HelseIdEnterpriseCertificateSecretHandler : SecretHandlerBase
     {
         private X509SecurityKey _x509SecurityKey;
 
-        public override string GenerateClientAssertion => ClientAssertion.Generate(configAuth.ClientId, configAuth.Authority, _x509SecurityKey);
-
-        public override void AddSecretConfiguration(IHelseIdWebKonfigurasjon configAuth, OpenIdConnectOptions options)
+        public HelseIdEnterpriseCertificateSecretHandler(IHelseIdWebKonfigurasjon helseIdWebKonfigurasjon) : base(helseIdWebKonfigurasjon)
         {
-            this.configAuth = configAuth;
-            var secretParts = configAuth.ClientSecret.Split(':');
+            var secretParts = ConfigAuth.ClientSecret.Split(':');
             if (secretParts.Length != 2)
             {
-                throw new InvalidEnterpriseCertificateSecretException(configAuth.ClientSecret);
+                throw new InvalidEnterpriseCertificateSecretException(ConfigAuth.ClientSecret);
             }
 
             var storeLocation = (StoreLocation)Enum.Parse(typeof(StoreLocation), secretParts[0]);
-            var thumprint = secretParts[1];
+            var thumbprint = secretParts[1];
 
             var store = new X509Store(storeLocation);
             store.Open(OpenFlags.ReadOnly);
 
-            var certificates = store.Certificates.Find(X509FindType.FindByThumbprint, thumprint, true);
+            var certificates = store.Certificates.Find(X509FindType.FindByThumbprint, thumbprint, true);
 
             if (certificates.Count == 0)
             {
-                throw new Exception($"No certificate with thumbprint {options.ClientSecret} found in store LocalMachine");
+                throw new Exception($"No certificate with thumbprint {ConfigAuth.ClientSecret} found in store LocalMachine");
```

Comment:
@OsirisTerje @martinmine Blir dette riktig?

Reply:
Tror denne er grei, dette er jo Enterprise handleren, og ikke brukt, så vi fjerner vel denne i neste runde

Reply:
ClientSecret i dette tilfellet er en JWT i ConfigAuth, og semantisk sett så blir exception-meldingen misvisende, tror dette er en mer beskrivende feilmelding av hva som skjer, hva tenker du?
```suggestion
                throw new Exception($"No certificate with thumbprint {thumbprint} found in store LocalMachine");
```

Reply:
Det ser mer riktig ut. Men som Terje sier forsvinner dette mest sannsynlig.